libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go

```diff
@@ -629,7 +629,6 @@ func (dhtPeer *DHTPeer) handleDelegateService(ready *sync.WaitGroup) {
 				go dhtPeer.handleNewDelegationConnection(conn)
 			}
 		case <-dhtPeer.closing:
-			break
```

Comment:
it's needed to exit connection loop
but you can set `err` here. so it will exit by err